### DISCUSSION TITLE
Can I add a navigation bar to my Quarto website with theme set to none?

### DISCUSSION THREAD

**Discussion by WillemSleegers** [quarto-dev/quarto-cli#8558] [Q&A] [answered]
### Description

Is it possible to set theme to none and still get a navigation bar, whose styling I can then set myself using CSS?

**Comment by mcanouil** [comment]
See <https://quarto.org/docs/output-formats/html-themes.html>.

**Reply by WillemSleegers** [reply]
Can you elaborate because I don't think it answers my question. 

If I set the theme to none (or pandoc); my navigation bar disappears.

**Reply by mcanouil** [reply]
Did you read the documentation?
A navbar cannot exist without (S)CSS, in this case Bootstrap.

**Reply by WillemSleegers** [reply]
Yes I did and I don't see that mentioned.

**Reply by mcanouil** [reply]
> theme: none    # no theme css added to document

From <https://quarto.org/docs/output-formats/html-themes.html#overview>.

> You can naturally also create an entirely custom theme and provide only that (in this case you will inherit from the default Bootstrap theme):
>
> ```yaml
> theme: custom.scss
> ```

From <https://quarto.org/docs/output-formats/html-themes.html#custom-themes>.

Quarto HTML format/projects is built using Bootstrap. Disabling Bootstrap means to disable all features relying on it.

**Reply by WillemSleegers** [reply]
Yes, and that does not say that the navigation bar is entirely removed. I think it's much more obvious to expect that the _styling_ of the navigation bar is removed; not the navigation bar itself.

> Quarto HTML format/projects is built using Bootstrap. Disabling Bootstrap means to disable all features relying on it.

Apparently; although it's not clear to everyone which features rely on Bootstrap and which one's don't.

**Reply by mcanouil** [reply]
FYI, all this has been discussed in various discussions:
- https://github.com/quarto-dev/quarto-cli/discussions/8144
- https://github.com/quarto-dev/quarto-cli/discussions/6812
- https://github.com/quarto-dev/quarto-cli/discussions/8259

Discussions about the navbar and theming:
- https://github.com/quarto-dev/quarto-cli/discussions?discussions_q=label%3Athemes+-label%3Arevealjs+-label%3Atypst+navbar

**Reply by mcanouil** [reply]
> Yes, and that does not say that the navigation bar is entirely removed. I think it's much more obvious to expect that the styling of the navigation bar is removed; not the navigation bar itself.

The CSS class makes it a navbar, otherwise it's just an html div/nav.
Since you'll have no way to put back CSS classes. it's not created at all when there is no theme.

<img width="1624" alt="image" src="https://github.com/quarto-dev/quarto-cli/assets/8896044/6ef112a8-e4e7-447c-8c5c-c6e168a1b973">

**Reply by WillemSleegers** [reply]
Yeah I'm not finding your responses particularly helpful; you could have just said no right at the start and mention that it's because it relies on Bootstrap, instead of mainly providing links.

**Reply by WillemSleegers** [reply]
>The CSS class makes it a navbar, otherwise it's just an html div/nav.

Sure but apparently the div/nav is removed. They could just leave that in but remove the class.

**Reply by mcanouil** [reply]
> Sure but apparently the div/nav is removed. They could just leave that in but remove the class.

And how would you style the navbar then? Providing a CSS file won't work.

> Yeah I'm not finding your responses particularly helpful; you could have just said no right at the start and mention that it's because it relies on Bootstrap, instead of mainly providing links.

You wanted to have the navbar and use custom (S)CSS.
That's what the "custom themes" section is about: <https://quarto.org/docs/output-formats/html-themes.html#custom-themes>

**Reply by WillemSleegers** [reply]
>And how would you style the navbar then? Providing a CSS file won't work.

They could make it a nav element so it can be targeted with CSS; or they can leave in the class name but don't load Bootstrap. The current behavior of removing it entirely is weird and seems like an accidental effect of how things are structured rather than an intentional choice.

>You wanted to have the navbar and use custom (S)CSS.

I wanted to have the navbar with theme set to none.

**Reply by mcanouil** [reply]
> I wanted to have the navbar with theme set to none.

To be clear, the theme is not set to the navbar, it's for the entire document/project.

> They could make it a nav element so it can be targeted with CSS; or they can leave in the class name but don't load Bootstrap. The current behavior of removing it entirely is weird and seems like an accidental effect of how things are structured rather than an intentional choice.

I strongly believe that's a design choice which to me makes sense (I am not a Quarto developer).
CC @dragonstyle 

> they can leave in the class name but don't load Bootstrap.

Class names are tied up to the CSS framework, it would makes no sense to keep the names and not use the framework.
Setting "neutral" names is not a viable option, because it basically means to create a new CSS framework.

**Reply by dragonstyle** [reply, marked as answer]
That definitely seems like a reasonable goal, but `theme: none` isn't really all that close to this right now. The entire website envelope is connected to theme, which includes note merely the navbar, but all the other HTML scaffolding.  As @mcanouil noted, the structure of the HTML itself and the attributes like classes are all oriented to support bootstrap. 

I think the right long term direction is actually to emit an intermediary form of HTML which semantically represents the simple structure of the site (e.g. navbars, sidebars, footers, etc) and then have a followup pass which applies bootstrap (or other HTML frameworks) to the semantic HTML but restructuring the DOM and injecting attributes. (This is basically how Dashboards work now - we actually have a semantic representation which we then decorate to take advantage of `bslib` rendering). This would mean we could have a mode that simply didn't apply bootstrap and left the semantic HTML as is, for users to style as they see fit.

This is a great feature, but is likely going to be pretty complicated given how much code we have written around website themes. I would love this and think this is the correct long term direction for Quarto, but it's a _big_ bite and not something that is likely to happen in the near term.

**Reply by WillemSleegers** [reply]
That's what I was afraid of, but glad to hear there's the motivation to potentially address it in the future!

**Reply by cscheid** [reply]
Let me temper everyone's expectations: we have no plan to address this in the next 12-24 months at the least. @dragonstyle might be understating the extent to which it would be a massive restructuring of how theming works in Quarto websites.

**Reply by dragonstyle** [reply]
I put italics on big! 🥇
